timely_beliefs/beliefs/classes.py

```diff
@@ -175,37 +197,102 @@ class BeliefsDataFrame(DataFrame):
     def _constructor(self):
         return BeliefsDataFrame
 
+    def __init__(self, *args, **kwargs):
+        """Initialise a multi-index DataFrame with beliefs about a unique sensor."""
+
+        # Obtain parameters that are specific to our DataFrame subclass
+        sensor: Sensor = kwargs.pop("sensor", None)
+        beliefs: List[TimedBelief] = kwargs.pop("beliefs", None)
+
+        # Use our constructor if initialising from a previous DataFrame (e.g. when slicing),
+        # copying the Sensor metadata
+        if beliefs is None:
+            super().__init__(*args, **kwargs)
+            return
+
+        # Define our columns and indices
+        columns = ["event_value"]
+        indices = ["event_start", "belief_time", "source", "belief_percentile"]
+
+        # Call the pandas DataFrame constructor with the right input
+        kwargs["columns"] = columns
+        if beliefs is not None:
+            kwargs["data"] = [[getattr(i, j) for j in columns] for i in beliefs]
+            kwargs["index"] = pd.MultiIndex.from_tuples(
+                [[getattr(i, j) for j in indices] for i in beliefs], names=indices
+            )
+        else:
+            kwargs["index"] = pd.MultiIndex(
+                levels=[[] for _ in indices],
+                labels=[[] for _ in indices],
+                names=indices,
+            )
+        super().__init__(*args, **kwargs)
+
+        # Set the Sensor metadata (including timing properties of the sensor)
+        self.sensor = sensor
+
+    def add_from_time_series(self,
+        series: pd.Series, source: BeliefSource, horizon: timedelta
+    ):
+        """Add beliefs from time series entries into this TimedBeliefsDataFrame.
+        Append makes no sense with this class atm."""
```

Comment:
Maybe `append()` does work now with the new code on master?

Reply:
It does work, but it also cannot be an inplace operation. I still made it possible via the BeliefsDataFrame class.